src/main/java/org/sonarqube/gradle/SonarQubePlugin.java

```diff
@@ -88,13 +92,36 @@ private void configureTask(SonarQubeTask sonarQubeTask, Project project, Map<Str
     conventionMapping.map("properties", () -> new SonarPropertyComputer(actionBroadcastMap, project)
       .computeSonarProperties());
 
-    Callable<Iterable<? extends Task>> testTasks = () -> project.getAllprojects().stream()
+    sonarQubeTask.mustRunAfter(getJavaTestTasks(project));
+    sonarQubeTask.dependsOn(getJavaCompileTasks(project));
+    sonarQubeTask.mustRunAfter(getJacocoTasks(project));
+    sonarQubeTask.dependsOn(getAndroidCompileTasks(project));
+  }
+
+  private static Callable<Iterable<? extends Task>> getJacocoTasks(Project project) {
+    return () -> project.getAllprojects().stream()
+      .filter(p -> p.getPlugins().hasPlugin(JacocoPlugin.class) && !p.getExtensions().getByType(SonarQubeExtension.class).isSkipProject())
+      .map(p -> p.getTasks().withType(JacocoReport.class))
+      .flatMap(Collection::stream)
+      .collect(Collectors.toList());
+  }
+
+  private static Callable<Iterable<? extends Task>> getJavaTestTasks(Project project) {
+    return () -> project.getAllprojects().stream()
       .filter(p -> p.getPlugins().hasPlugin(JavaPlugin.class) && !p.getExtensions().getByType(SonarQubeExtension.class).isSkipProject())
       .map(p -> p.getTasks().getByName(JavaPlugin.TEST_TASK_NAME))
       .collect(Collectors.toList());
-    sonarQubeTask.dependsOn(testTasks);
+  }
 
-    Callable<Iterable<? extends Task>> compileTasks = () -> project.getAllprojects().stream()
+  private static Callable<Iterable<? extends Task>> getJavaCompileTasks(Project project) {
+    return () -> project.getAllprojects().stream()
+      .filter(p -> p.getPlugins().hasPlugin(JavaPlugin.class) && !p.getExtensions().getByType(SonarQubeExtension.class).isSkipProject())
```

Comment:
The project (and extension) might yet be unconfigured at the time this code is executed.